### PR TITLE
Make Hazelcast configurable

### LINKED ISSUE #1392: Make the Hazelcast settings configurable
# Task Description

Currently, the Hazelcast configuration is hard-coded in the `org.carlspring.strongbox.config.hazelcast.HazelcastConfiguration`. We need to be able to configure this via the `application.yaml` instead.

# Tasks

The following tasks will need to be carried out:
* [ ] Make the Hazelcast settings configurable via the `application.yaml`.
* [ ] ~~Make the Hazelcast settings configurable via the controller.~~ (this is currently not possible)

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by cbrianhill**
# Pull Request Description

This pull request closes #1392

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [ ] No

### CODE REVIEW

File: strongbox-commons/src/main/resources/application.yaml [hunk 1]
```diff
@@ -62,4 +62,31 @@ management:
       exposure:
         include: health,info,beans,metrics,trace,scheduledtasks,threaddump
       base-path: /api/monitoring
-
+cacheManagerConfiguration:
+  groupConfig:
+    name: strongbox
+    password: password
```
Comment (carlspring) [inline strongbox-commons/src/main/resources/application.yaml]: Could we have this as a non-plain-text value? @strongbox/core-developers : Any suggestions?

And... perhaps the `application.yaml` should only be viewable by the owner (we can control this in the `strongbox-distribution`'s assembly descriptor).

**Comment by carlspring** [comment]
@cbrianhill : Thanks for this pull!

@strongbox/core-developers : Could you please, review this? Thanks! :)

File: strongbox-commons/src/main/resources/application.yaml (hunk 1 above)
Comment (sbespalov) [reply strongbox-commons/src/main/resources/application.yaml]: > Could we have this as a non-plain-text value? @strongbox/core-developers : Any suggestions?

we can try to research maybe this is possible to use certificate for this

> And... perhaps the application.yaml should only be viewable by the owner (we can control this in the strongbox-distribution's assembly descriptor).

AFAIK this `application.yaml` is under classpath, so you can't directly change or read it

File: strongbox-commons/src/main/resources/application.yaml [hunk 2]
```diff
@@ -62,4 +62,31 @@ management:
...
+  enableMulticastConfig: true
```
Comment (sbespalov) [inline strongbox-commons/src/main/resources/application.yaml]: @cbrianhill multicast should be disabled by default

**Comment by sbespalov** [comment]
BTW, @carlspring 

should we expose all of this properties as env variables like we already have for [Application Properties](https://strongbox.github.io/user-guide/application-properties.html#application-properties)?

File: strongbox-commons/src/main/resources/application.yaml (hunk 1 above)
Comment (cbrianhill) [reply strongbox-commons/src/main/resources/application.yaml]: What do you think about `jasypt-spring-boot` for encrypting these properties (https://github.com/ulisesbocchio/jasypt-spring-boot)?

The `application.yaml` actually is baked into a jar file (the `strongbox-commons` JAR file which is then included in the spring-boot uber-jar.  Would it be a good convention for the uber-jar to be readable only by the owner?

File: strongbox-commons/src/main/resources/application.yaml (hunk 2 above)
Comment (cbrianhill) [reply strongbox-commons/src/main/resources/application.yaml]: Good catch, will disable this.  I'd flipped it for local testing.

**Comment by cbrianhill** [comment]
> BTW, @carlspring
> 
> should we expose all of this properties as env variables like we already have for [Application Properties](https://strongbox.github.io/user-guide/application-properties.html#application-properties)?

Yeah, I considered this, please let me know your thoughts.  It would be quite a few environment variables, but happy to wire them up.

File: strongbox-commons/src/main/resources/application.yaml (hunk 1 above)
Comment (carlspring) [reply strongbox-commons/src/main/resources/application.yaml]: @cbrianhill ,

> The application.yaml actually is baked into a jar file (the strongbox-commons JAR file which is then included in the spring-boot uber-jar. Would it be a good convention for the uber-jar to be readable only by the owner?

We actually need to do something else... 

At the moment, all the properties are being loaded from the classpath. We should check, if they are actually available on the file system and load them from there first and only fallback to the classpath (from within the jar).

**Comment by carlspring** [comment]
@sbespalov ,

> should we expose all of this properties as env variables like we already have for [Application Properties](https://strongbox.github.io/user-guide/application-properties.html#application-properties)?

If you're referring to the ones from the `application.yaml`, I don't think this would be a reasonable idea.

**Comment by steve-todorov** [comment]
I agree with @carlspring - exposing this configuration as env vars would be a bit of an overkill and a possible security risk. The password is currently in plain-text and even if it weren't exposing the password as an env variable is probably not a very good idea.

**Comment by sbespalov** [comment]
ok, thanks for clarifying
in general this PR seems very good, thanks @cbrianhill

**Comment by carlspring** [comment]
Thanks for this fix, @cbrianhill ! :)

**Pull request merged**